src/compas/geometry/collections/collection.py

```diff
@@ -2,15 +2,240 @@
 from __future__ import absolute_import
 from __future__ import division
 
-__all__ = ['Collection']
+from compas.data import Data
+from compas.data.encoders import cls_from_dtype
 
 
-class Collection(object):
+class Collection(Data):
+    """A collection of data is a mutable sequence of COMPAS data objects of the same type.
+
+    Parameters
+    ----------
+    itype : Type[:class:`compas.geometry.Geometry`]
+        The type of very item in the collection.
+    items : sequence[:class:`compas.geometry.Geometry`], optional
+        The items of the collection.
+        These items can be instances of COMPAS geometry objects,
+        or their Python built-in equivalent.
+
+    Attributes
+    ----------
+    items : list[:class:`compas.geometry.Geometry`]
+        The type of each item is the same as :attr:`itype`.
+
+    Examples
+    --------
+    >>> from compas.geometry import Pointcloud, Line, Collection
+    >>> from compas.utilities import pairwise
+
+    Construct from list of items.
+
+    >>> lines = []
+    >>> for a, b in pairwise(Pointcloud.from_bounds(8, 5, 3, 31)):
+    ...     lines.append([a, b])
+    ...
+    >>> collection = Collection(Line, lines)
+    >>> collection
+    '<Collection of 30 Line>'
+
+    Construct with loop.
+
+    >>> collection = Collection(Line)
+    >>> for a, b in pairwise(Pointcloud.from_bounds(8, 5, 3, 31)):
+    ...     collection.append([a, b])
+    ...
+    >>> collection
+    '<Collection of 30 Line>'
+
+    """
 
     __slots__ = ['_items']
 
-    def transform(self, X):
+    def __init__(self, itype, items=None, **kwargs):
+        super(Collection, self).__init__(**kwargs)
+        self._itype = itype
+        self._items = []
+        self.items = items
+
+    # ==========================================================================
+    # items
+    # ==========================================================================
+
+    @property
+    def itype(self):
+        return self._itype
+
+    @property
+    def items(self):
+        return self._items
+
+    @items.setter
+    def items(self, items):
+        if not items:
+            return
+        if not self._itype:
+            raise Exception("Item type not set.")
+        self._items = []
+        for item in items:
+            self.add(item)
+
+    # ==========================================================================
+    # data
+    # ==========================================================================
+
+    @property
+    def DATASCHEMA(self):
+        """schema.Schema : Schema of the data representation."""
+        from schema import Schema
+        return Schema({
+            'itype': '.../...',
+            'items': lambda item: 'validate schema?'
```

Comment:
This definitely needs an `itype` before merging, as well as something in the `items` stuff.

Reply:
Actually... what is `itype`? :P `item_type` I assume, but it's treated used like the `dtype` of the data serialization framework with the same 2-level contraint, which I'm not sure makes so much sense here.